src/node_jsvmapi.h

```diff
@@ -116,80 +116,99 @@ enum napi_valuetype {
   napi_function,
 };
 
-// Environment
-NODE_EXTERN napi_env napi_get_current_env();
+enum napi_errorvalue {
```

Comment:
@mhdawson In my (admittedly limited) experience reviewing native module code, it's fairly unusual for the native module to care whether or not there is a pending exception. Most of the time, they can just let it remain pending so that it gets thrown whenever the native call returns to JS. Therefore, I think it might not be worth the added complexity to have status codes that are a combination of two things instead of a simple enum.